golem_sci/blockshelper.py

```diff
@@ -6,10 +6,11 @@ class BlocksHelper:
     def __init__(self, sci: SmartContractsInterface) -> None:
         self._sci = sci
 
-    def get_first_block_after(self, timestamp: int) -> Block:
+    def get_latest_existing_block_at(self, timestamp: int) -> Block:
```

Comment:
I somewhat interpret this name as "give me the latest block that existed at time `timestamp`" which isn't correct.
I can't think of a better name though.